src/components/SourceDetail/AvailabilityChecker.js

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useDispatch } from 'react-redux';
+
+import RedoIcon from '@patternfly/react-icons/dist/js/icons/redo-icon';
+import { Button } from '@patternfly/react-core/dist/js/components/Button/Button';
+import { Spinner } from '@patternfly/react-core/dist/js/components/Spinner/Spinner';
+
+import { useSource } from '../../hooks/useSource';
+import checkSourceStatus from '../../api/checkSourceStatus';
+import { addMessage } from '../../redux/sources/actions';
+
+const AvailabilityChecker = () => {
+  const source = useSource();
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState(false);
+
+  return (
+    <Button
+      variant="plain"
+      aria-label={intl.formatMessage({ id: 'sources.checkavailability', defaultMessage: 'Check source availability' })}
+      onClick={async () => {
+        setLoading(true);
+        await checkSourceStatus(source.id);
+        setLoading(false);
+        dispatch(
+          addMessage(
+            intl.formatMessage({
+              id: 'sources.checkavailability.notificationTitle',
+              defaultMessage: 'Request to check source status was sent',
+            }),
+            'info',
+            intl.formatMessage({
+              id: 'sources.checkavailability.notificationDescription',
+              defaultMessage: 'Check this page later for updates',
+            })
+          )
+        );
+      }}
+      isDisabled={loading}
+    >
+      {!loading && <RedoIcon />}
```

Comment:
You should use the `icon` prop if you want icon buttons.

Reply:
No, plain buttons need icons as children. See https://www.patternfly.org/v4/components/button

Reply:
:face_with_head_bandage: consistent API eh?